src/languages/haskell.js

```diff
@@ -62,6 +62,42 @@ export default function(hljs) {
     contains: LIST.contains
   };
 
+  /* See:
+
+     - https://www.haskell.org/onlinereport/lexemes.html
+     - https://downloads.haskell.org/ghc/9.0.1/docs/html/users_guide/exts/binary_literals.html
+     - https://downloads.haskell.org/ghc/9.0.1/docs/html/users_guide/exts/numeric_underscores.html
+     - https://downloads.haskell.org/ghc/9.0.1/docs/html/users_guide/exts/hex_float_literals.html
+
+  */
+  const decimalDigits = '([0-9]_*)+';
+  const hexDigits = '([0-9a-gA-F]_*)+';
```

Comment:
a-g?

Reply:
Woops, leftover of a debugging attempt :facepalm:. Fixed!